src/main/java/com/amihaiemil/zold/WaitForWallet.java

```diff
@@ -87,25 +93,92 @@ public Wallet handleResponse(final HttpResponse httpResponse)
             );
         } else {
             try {
-                Thread.sleep(1000);
-                final URIBuilder uri = new URIBuilder(
-                    this.baseUri.toString() + "/job"
-                ).addParameter("id", jobId);
-                final HttpGet status = new HttpGet(uri.build());
-                this.client.execute(
-                    status,
-                    new MatchStatus(
-                        status.getURI(),
-                        HttpStatus.SC_OK
-                    )
-                );
+                final Duration start = Duration.ofNanos(System.nanoTime());
+
+                do {
+                    Thread.sleep(1000);
+
+                    if (this.jobFinished(jobId)) {
+                        break;
+                    }
+
+                    if (this.hasNoTimeToWait(start)) {
+                        throw new IllegalStateException(
+                                "Can't wait for job " + jobId + " during "
+                                + this.timeToWait.toMinutes() + " minutes");
+                    }
+
+                } while (true);
+
                 return new RtWallet(this.client, this.baseUri);
             } catch (final InterruptedException | URISyntaxException ex) {
                 throw new IllegalStateException(
```

Comment:
@OleksandrSankin `InterruptedException` is not handled correctly here, catch block should call `Thread.currentThread().interrupt()` if method can handle this kind of exception or rethrow it as-is if can't, see https://stackoverflow.com/q/3976344/1723695

I don't think that throwing `IllegalStateException` when thread was interrupted is a right design.
// cc @amihaiemil

Reply:
@g4s8 we'll fix that in #39, thanks :)